docs/dev/development/routine-maintenance.md

```diff
@@ -0,0 +1,43 @@
+---
+title: Databend routine maintenance
+---
+
+## What is Databend routine maintenance?
+
+Routine maintenance of Databend is currently performed roughly every 30 days. This is usually done at the beginning of the month.
+
+The main tasks are to **update the toolchain** and **upgrade the dependencies**.
+
+## Update the toolchain
+
+Databend will choose a relatively new nightly version for the build/test/bench.
+
+The steps to update toolchain are as follows:
+
+- edit `rust-toolchain.toml` to update the `channel` field.
+  - Usually updated to the current date.
+- Run `make lint` to make sure clippy is happy.
+  - clippy is not entirely correct. If necessary, you can use `#[allow(clippy::xxx)]` to skip some of the rules.
+  - If possible, please add a comment to clarify.
+- Run `make test` to check the correctness.
+
+## Upgrade the dependencies
+
+Databend involves nearly a thousand third-party dependencies from sources such as [github](https://github.com/) and [crates.io](https://crates.io/).
+
+Currently, the dependency upgrade process is divided into the following steps:
+
+- Run `cargo upgrade --workspace` to upgrade dependencies in your Cargo.toml to their latest versions.
+  - [cargo-edit](https://crates.io/crates/cargo-edit) crate needs to be installed.
+  - Please remember to perform step 4.
```

Comment:
At first glance, I don't know what `step 4` means, maybe it's better to use:

```
- Step 1: Run `cargo upgrade --workspace` ....
- Step 2: Check the
- Step 3: Update dependencies
- Step 4: Run `make lint` and `make test`
```

Btw, we don't need to notify readers everywhere (they just read step 1...)

We can use `**Please make sure all check passed**` in step 4 to mark it for readers.

Reply:
got